lib/AST/Decl.cpp

```diff
@@ -1031,7 +1031,7 @@ unsigned PatternBindingDecl::getPatternEntryIndexForVarDecl(const VarDecl *VD) c
     ++Result;
   }
   
-  assert(0 && "PatternBindingDecl doesn't bind the specified VarDecl!");
+  llvm_unreachable("PatternBindingDecl doesn't bind the specified VarDecl!");
```

Comment:
So .. some portion of these changes were to deal with the fact that I was seeing assert(0 && "foo") generating an unreachable-code warning too, which really makes no sense; so I rechecked in more detail and .. it seems some portion of the mess of unreachability-warnings was ccache interference. I'll recheck these tomorrow.

(The unused-vars and calling-convention stuff is, I think, more certainly real)